compute_endpoint/globus_compute_endpoint/endpoint/config/utils.py

```diff
@@ -182,6 +183,16 @@ def _sanitize_user_opts(data):
         raise ValueError(f"{type(data).__name__} is not a valid user option type")
 
 
+def _shell_escape(val: str):
+    """Returns a shell-escaped version of the argument"""
+    # We need to escape the command before serializing into JSON
+    # because PyYAML will strip the surrounding quotes when the
+    # YAML content is loaded
+    val = json.loads(val)
+    esc_val = shlex.quote(val)
```

Comment:
Is there any value in verifying that the passed data structure is, in fact, a string?  And deserializes to a string?

Perhaps:
```python
if not isinstance(val, str):
    raise TypeError(...)

loaded = json.loads(val)
if not isinstance(loaded, str):
    raise ValueError(...)

return json.dumps(shlex.quote(loaded))
```

Reply:
I like that, but I'd prefer to return the original arg value when it's not a string rather than raise an exception.